tests/kafkatest/tests/core/reassign_partitions_test.py

```diff
@@ -176,7 +176,10 @@ def test_reassign_partitions(self, bounce_brokers, reassign_from_offset_zero, me
         self.producer = VerifiableProducer(self.test_context, self.num_producers,
                                            self.kafka, self.topic,
                                            throughput=self.producer_throughput,
-                                           enable_idempotence=True)
+                                           enable_idempotence=True,
+                                           # To avoid the reassignment behavior being affected by `partitioner.ignore.keys=true`,
+                                           # we set a key for the message to ensure the message order is preserved within the partition.
```

Comment:
We need this change due to the test behavior, so it would be helpful to emphasize that point.
```
This test expects that each partition can receive the record, so using a non-null key helps distribute the records more randomly.
```